akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpHeaderParser.scala

```diff
@@ -438,9 +440,18 @@ private[http] object HttpHeaderParser {
     "Cache-Control: no-cache",
     "Expect: 100-continue")
 
-  def apply(settings: HttpHeaderParser.Settings, log: LoggingAdapter)(onIllegalHeader: ErrorInfo ⇒ Unit = info ⇒ throw IllegalHeaderException(info)) =
+  def apply(settings: HttpHeaderParser.Settings, log: LoggingAdapter)(onIllegalHeader: ErrorInfo ⇒ Unit = defaultIllegalHeaderHandler(settings, log)) =
```

Comment:
I went over the usages of `HttpHeaderParser.apply()`  and there's one of two things that happens every time with regards to the illegal header handling:
- the call defers to the default value ([`RequestParserSpec 305`](https://github.com/akka/akka-http/blob/master/akka-http-core/src/test/scala/akka/http/impl/engine/parsing/RequestParserSpec.scala#L305), [`RequestParserSpec 587`](https://github.com/akka/akka-http/blob/master/akka-http-core/src/test/scala/akka/http/impl/engine/parsing/RequestParserSpec.scala#L587), [`WebSocketClientBlueprint`](https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/impl/engine/ws/WebSocketClientBlueprint.scala#L65), [`ResponseParserSpec`](https://github.com/akka/akka-http/blob/master/akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala#L326), and `HttpHeaderParserSpec` where I just added a usage)
- a variation of the same function is passed, which looks at the parser settings and either warns about the illegal header or not and then moves on ([`OutgoingConnectionBlueprint`](https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala#L89-L92), [`BodyPartParser`](https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/impl/engine/parsing/BodyPartParser.scala#L62-L64), [`HttpServerBluePrint`](https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala#L212-L215), [`Http2Blueprint`](https://github.com/akka/akka-http/blob/master/akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala#L82-L85))

Which kinda begs the question of why the option to pass that function in exists at all. In my opinion either all of the usages currently passing in a function should defer to the default, or the option to pass a custom function for illegal header handling should be removed altogether. Thoughts?